### ISSUE TITLE
1.5.12 : X20 hang (touch screen & sound) without loosing the control

### ISSUE THREAD

**Issue by filduch** [FrSkyRC/ETHOS-Feedback-Community#4376]
Since the upgrade of Ethos version 1.5.11 to 1.5.12, several times my X20 was blocked without losing flight control. The sound had disappeared as well as the touch part of the screen. Disconnecting the battery and putting it back in solved the problem.
I would be curious to know if I am the only one to have noticed this malfunction ?

**Comment by filduch** [comment]
I now have the problem regularly. I noticed that using the touchscreen still works but with a reaction time of about 5 seconds. I have the impression that the X20 is overloaded and I wonder how to know if it is indeed.
It would be possible that the script using:
local lipoSensor = system.getSource({name=widget.value:name(), category=CATEGORY_TELEMETRY_SENSOR })
local numCells = system.getSource({member = lipoSensor:member(), category=CATEGORY_TELEMETRY_SENSOR, options=OPTION_CELL_COUNT}) and
local cellSensor = system.getSource({member = lipoSensor:member(), category=CATEGORY_TELEMETRY_SENSOR, options=OPTION_CELL_INDEX(cell)}) could be the code causing the overload. 
I will try to do without the script to see if it is responsible for the problem.

**Comment by strgaltdel** [comment]
declaring sources can't overload the system (when used in the right way)

but this was a similar issue, you may check it:
#4355

**Comment by bsongis-frsky** [comment]
5 seconds seems slow, are you sure? Try for example to press a key and see how many seconds are needed before it is taken into account.

Then I suggest you remove your scripts directory and add the scripts one by one to find which one is too slow

Also `system.getSource()` from a name may be slow, as it searches in the whole system if there is something with this name ...

**Comment by filduch** [comment]
The problem does not come from the active luas (I disabled them all and the problem was still there).
Recently, the touch screen and sound were still missing and I noticed that only the time was updated every 5 seconds. I also noticed that access to the SD card was no longer possible.
I have now changed the SD card and have not yet been able to reproduce the problem. I will see if the problem does not appear in the near future to be sure that the SD card was the cause.
